data/battle_scripts_1.s

```diff
@@ -2346,6 +2349,16 @@ BattleScript_EffectSimpleBeam::
 	setabilitysimple BS_TARGET, BattleScript_ButItFailed
 	attackanimation
 	waitanimation
+.if B_ABILITY_POP_UP == TRUE
+	setbyte sFIXED_ABILITY_POPUP, TRUE
+	showabilitypopup BS_TARGET
+	pause 60
+	sethword sABILITY_OVERWRITE, 0
+	updateabilitypopup BS_TARGET
+	pause 20
+	destroyabilitypopup
+	pause 40
+.endif
```

Comment:
do you think it would make sense to make a script for those to reduce duplicate scripts?

Reply:
Added `BattleScript_AbilityPopupOverwriteThenNormal` which requires `gBattlerAbility` to be set to the battler, to be within `.if ABILITY_POPUP == TRUE` and `sABILITY_OVERWRITE` to be set before being called.

Reply:
I also added some missing `recordability` calls.

Reply:
> Added `BattleScript_AbilityPopupOverwriteThenNormal` which requires `gBattlerAbility` to be set to the battler, to be within `.if B_ABILITY_POPUP == TRUE` ... `.endif` and `sABILITY_OVERWRITE` to be set before being called.

Damn, I asked specifically to get rid of the preprocs :sweat_smile:

I let you decide if the change is worth it otherwise.